src/filter/mod.rs

```diff
@@ -62,15 +63,25 @@ pub trait Filter {
     fn run(
         &mut self,
         recv: hopper::Receiver<metric::Event>,
+        sources: std::vec::Vec<std::string::String>,
```

Comment:
I do believe you can make this just `Vec<String>`, no? There are some things auto-used.

Reply:
Shortened everywhere.

Along with String and Option.